### DISCUSSION TITLE
[Feature improvement] - El-input with mask

### DISCUSSION THREAD

**Discussion by ntgraph** [element-plus/element-plus#936] [Ideas]
Hi,
I wish there is a mask option in el-input, such as like this
:mask="####"

it can be number of digit, decimal number , certain specific character or number in the input, or Regex string which in turn is providing from data or computed.

Example:
<el-input mask="(###) ###-####" 

on screen, the user can only type number, and it display like this (543) 345-4534

**Comment by jw-foss** [comment]
Ok, so to me this is more like a formatting, when you give the format lilke (###) ###-####, using the regex or whatever to match it then replace the actual input with `(replaced) replaced - replaced,` something like this right?

**Comment by ntgraph** [comment]


**Reply by jw-foss** [reply]
I actually don't quite understand, if there is a example that I could take a look, that will be better.

**Reply by ntgraph** [reply]
Sorry, my last post were not clear, please look this site, I am suggesting the el-input to have a mask.  Instead of accepting all input values, we can apply a mask, then the input must follow that template
see link below:
https://vuejs-tips.github.io/vue-the-mask/

Thanks,

**Comment by euthiago** [comment]
Some examples:
https://github.com/nosir/cleave.js/
https://github.com/beholdr/maska
And, as already shown, https://vuejs-tips.github.io/vue-the-mask/
Currently integrating with those projects is quite hard.
I managed to make cleave.js and maska work, but feels like a lot of work for something that should be more seamlessly integrated.
The most important part is allowing a masked value for the input without losing the v-model raw value binding.

**Comment by earlAchromatic** [comment]
A really good example of a component library that does this well is primevue. See [https://www.primefaces.org/primevue/showcase/#/inputmask](https://www.primefaces.org/primevue/showcase/#/inputmask). But that lib is not in typescript.

This is a must-have feature for implementing basic forms. Is there any update on this feature being added?
